src/applications/claims-status/utils/useDatadogRum.js

```diff
@@ -0,0 +1,112 @@
+import { useEffect } from 'react';
+import { datadogRum } from '@datadog/browser-rum';
+import { datadogLogs } from '@datadog/browser-logs';
+
+import environment from '@department-of-veterans-affairs/platform-utilities/environment';
+import { useFeatureToggle } from '~/platform/utilities/feature-toggles';
+
+// https://docs.datadoghq.com/real_user_monitoring/browser/#configuration
+const defaultRumSettings = {
+  // Custom settings
+  // applicationId: '{UUID}',
+  // clientToken: 'pubb{ID}',
+  // service: 'benefits-{app name}',
+  // version: '1.0.0',
+
+  // Default settings
+  site: 'ddog-gov.com',
+  // see src/site/constants/vsp-environments.js for defaults
+  env: environment.vspEnvironment(), // 'production'
+  sessionSampleRate: 5,
+  sessionReplaySampleRate: 20,
+  trackInteractions: true,
+  trackUserInteractions: true,
+  trackFrustrations: true,
+  trackResources: true,
+  trackLongTasks: true,
+  defaultPrivacyLevel: 'mask-user-input',
+};
+
+// Initialize Datadog RUM directly, if not using a feature flag
+// Don't call this function if not logged in
+const initializeRealUserMonitoring = customRumSettings => {
+  // Prevent RUM from re-initializing the SDK OR running on local/CI environments.
+  if (
+    !environment.BASE_URL.includes('localhost') &&
+    !window.DD_RUM?.getInitConfiguration() &&
+    !window.Mocha
+  ) {
+    datadogRum.init({
+      ...defaultRumSettings,
+      ...customRumSettings,
+    });
+
+    // If sessionReplaySampleRate > 0, we need to manually start the recording
+    datadogRum.startSessionReplayRecording();
+  }
+};
+
+// https://docs.datadoghq.com/logs/log_collection/javascript/?tab=us#configuration
+const defaultLogSettings = {
+  // Custom settings
+  // applicationId: '{UUID}',
+  // clientToken: 'pubb{ID}',
+  // service: 'benefits-{app name}',
+  // version: '1.0.0',
+
+  // default settings
+  site: 'ddog-gov.com',
+  // see src/site/constants/vsp-environments.js for defaults
+  env: environment.vspEnvironment(), // 'production'
+  sessionSampleRate: 5,
+  forwardErrorsToLogs: true,
+  forwardConsoleLogs: ['error'],
+  forwardReports: [],
+  telemetrySampleRate: 20, // default 20
```

Comment:
I think we want both this and `sessionSampleRate` to be 100

Reply:
@jerekshoe No, we've been approved for different settings on CST. See Steve Alber's comment here: https://dsva.slack.com/archives/C04KHCT3ZMY/p1712607684997799?thread_ts=1712607291.510969&cid=C04KHCT3ZMY

Reply:
Yeah it looks like he only specified that for session replay stuff, not for logging. I'm pretty sure we need 100% sample rates for error logging if we are going to replace Sentry with this

Reply:
I see, that makes sense. Waiting on confirmation from Steve and then will update. Thanks @jerekshoe